pkg/controller/egress/controller.go

```diff
@@ -137,6 +139,23 @@ func NewEgressController(crdClient clientset.Interface,
 		}
 		return externalIPPools, nil
 	}})
+	// externalIPPoolIndex will be used to get all Egresses associated with a given EgressIP.
+	egressInformer.Informer().AddIndexers(cache.Indexers{egressIPIndex: func(obj interface{}) (strings []string, e error) {
```

Comment:
append the index func in the above call.

Reply:
Removed the restriction for sharing EgressIP. So as this index.